pkg/sql/sqlstats/ssprovider.go

```diff
@@ -183,6 +188,8 @@ type Provider interface {
 
 // RecordedStmtStats stores the statistics of a statement to be recorded.
 type RecordedStmtStats struct {
+	SessionID       uint128.Uint128
+	StatementID     uint128.Uint128
```

Comment:
We should probably move `ClusterWideID` out of the `sql` package so you can use it here. I'm not sure the best place to move it to, maybe a new sub-package in `pkg/sql/cluster` and call it `cluster.ID`? I have a hunch that @ajwerner has a thoughtful opinion on the matter. If you do make the change, I recommend doing it in a separate commit before this one.

Reply:
Talking live with @ajwerner, let's move it to `base`. We're thinking now about a better name, to distinguish it from `base.ClusterID`.

Reply:
Or `pkg/sql/clusterwide`, or `pkg/sql/clusterunique`, calling it `clusterunique.ID`.

Reply:
Underway in #80667.